snapd-glib/snapd-client.c

```diff
@@ -764,6 +772,17 @@ static GSocket *open_snapd_socket(const gchar *socket_path,
   return g_steal_pointer(&socket);
 }
 
+#ifdef BUILD_TESTS
```

Comment:
What about exposing the `open_snapd_socket` function in a private header instead?

So that we don't have to do the actual testing inside the main file

Reply:
Mmm... good point...

Reply:
Done.